readme/coding_style.md

```diff
@@ -0,0 +1,170 @@
+> Coding style is mostly enforced by a pre-commit hook that runs `eslint`. This hook is installed whenever running `yarn install` on any of the application directory. If for some reason the pre-commit hook didn't get installed, you can manually install it by running `yarn install` at the root of the repository.
+
+
+
+# Rules
+
+## Use TypeScript for new files
+
+> Even if you are **modifying** a file that was originally in JavaScript you should ideally convert it first to TypeScript before modifying it.
+
+### Creating a new `.ts` file
+
+Because the TypeScript compiler generates `.js` files, be sure to add these new `.js` files to `.eslintignore` and `.gitignore`.
+
+To do this,
+1. If the TypeScript compiler has already generated a `.js` file for the new `.ts` file, delete it.
+2. Run `yarn run updateIgnored` in the root directory of the project (or `yarn run postinstall`)
+
+
+## Filenames
+
+ * `camelCase.ts`: Files that export multiple things.
+   * Example: [`checkForUpdates.ts`](https://github.com/laurent22/joplin/blob/dev/packages/app-desktop/checkForUpdates.ts)
+ * `PascalCase.ts`: [Only if the file contains a single class, which is the default export.](https://github.com/laurent22/joplin/pull/6607#discussion_r906847156)
+ * `types.ts` or `fooTypes.ts`: [Shared type definitions](https://github.com/laurent22/joplin/pull/6607#discussion_r906847156)
+   * Example : [`types.ts`](https://github.com/laurent22/joplin/blob/dev/packages/server/src/utils/types.ts)
+
+
+## Use `camelCase` for `const`ants in new code
+
+**BAD:**
+```ts
+// Bad! Don't use in new code!
+const GRAVITY_ACCEL = 9.8;
+```
+
+**Good:**
+```ts
+const gravityAccel = 9.8;
+```
+
+
+## Indent using `tab`s
+
+> **VSCode**: In `vscode`, be sure to check whether new files are created with `tab` or `space` indentation! [Spaces can be converted to tabs using the command palette.](https://code.visualstudio.com/docs/editor/codebasics#_autodetection)
+
+
+## Avoid `==`
```

Comment:
This section could be called `Use strict equality`.

With that explanation maybe:

> Using strict equality ensures that the correct types are used throughout the code base.

Reply:
Is there an exception for comparing to `null` (to allow `undefined`, but to not allow `0`)?

(If a variable can be `undefined`, `null`, or have a value, is it best to use `if (variable) { ... }`, `if (variable != null) { ... }`, or `if (variable !== null && variable !== undefined) { ... }`? The solution might be to avoid such cases if possible.)